src/compiler/checker.ts

```diff
@@ -11900,23 +11900,23 @@ namespace ts {
                         Debug.fail("Unhandled object type " + Debug.formatObjectFlags(type.objectFlags));
                     }
                 }
-                else if (type.flags & TypeFlags.Union) {
-                    resolveUnionTypeMembers(type as UnionType);
+                else if (isUnionType(type)) {
+                    resolveUnionTypeMembers(type);
                 }
-                else if (type.flags & TypeFlags.Intersection) {
-                    resolveIntersectionTypeMembers(type as IntersectionType);
+                else if (isIntersectionType(type)) {
+                    resolveIntersectionTypeMembers(type);
                 }
                 else {
-                    Debug.fail("Unhandled type " + Debug.formatTypeFlags(type.flags));
+                    Debug.fail("Unhandled type " + Debug.formatTypeFlags((type as unknown as Type).flags));

```

Comment:
Why is this cast required? What is `type` here? `never`?

Reply:
Yeah, it's `never`. I did a quick search for the error message in the issue tracker and couldn't find anything.